### PR TITLE
heppy-side of factorized JECs

### PR DESCRIPTION
**Pull request by jpata**
As discussed with @arizzi, https://github.com/cbernet/cmssw/pull/78 was left out of https://github.com/vhbb/cmssw/pull/572  so we pull it in here.

### CODE REVIEW

**Comment by jpata** [comment]
Had to resolve conflicts with the `isHttSubjet` stuff (following commits by @gkasieczka)
~~~
6226a88c6c3373ded164c9c004159c9b61128e7a
65156d04e6f8effae725645d9676a6889e15fb9a
eb48e8a0e76e55e2902d8548323857d8e4caa683
5f027f93ce859502849d54ef2822ef4c499a42d0
~~~
in `JetReCalibrator.py`, I simply took the heppy version.

**Comment by jpata** [comment]
@arizzi can this be merged?

**Comment by arizzi** [comment]
looking into it

File: PhysicsTools/Heppy/python/physicsutils/JetReCalibrator.py [hunk 1]
```diff
@@ -97,21 +121,15 @@ def getCorrection(self,jet,rho,delta=0,metShift=[0,0],corrector=None,isHttSubjet
...
-        if True:
-            if not self.JetUncertainty: raise RuntimeError("Jet energy scale uncertainty shifts requested, but not available")
-            self.JetUncertainty.setJetEta(jet.eta())
-
-            if isHttSubjet:
-                self.JetUncertainty.setJetPt(corr * jet.pt())
-            else:
-                self.JetUncertainty.setJetPt(corr * jet.pt() * jet.rawFactor())
-
+        if delta != 0:
```
Comment (arizzi) [inline PhysicsTools/Heppy/python/physicsutils/JetReCalibrator.py:112]: mhh you reintroduced the "if delta!= 0" that I think *we* removed

File: PhysicsTools/Heppy/python/physicsutils/JetReCalibrator.py [hunk 2]
```diff
@@ -47,43 +56,58 @@ def __init__(self,globalTag,jetFlavour,doResidualJECs,jecPath,upToLevel=3,
...
-                    for i in [self.L2JetPar,self.L3JetPar]: self.vParL3.push_back(i)
-                    self.separateJetCorrectors["L2L3"] = ROOT.FactorizedJetCorrector(self.vParL3)
-                if doResidualJECs and calculateSeparateCorrections:
-                    self.vParL3Res = ROOT.vector(ROOT.JetCorrectorParameters)()
-                    for i in [self.L2JetPar,self.L3JetPar,self.ResJetPar]: self.vParL3Res.push_back(i)
-                    self.separateJetCorrectors["L2L3Res"] = ROOT.FactorizedJetCorrector(self.vParL3Res)
-
-    def getCorrection(self,jet,rho,delta=0,metShift=[0,0],corrector=None,isHttSubjet=False):
-        """Calculates the correction factor of a jet without modifying it
+            self.vParL1 = ROOT.vector(ROOT.JetCorrectorParameters)()
```
Comment (arizzi) [inline PhysicsTools/Heppy/python/physicsutils/JetReCalibrator.py:64]: you also removed the skipLevel1 option right?

Comment (jpata) [reply PhysicsTools/Heppy/python/physicsutils/JetReCalibrator.py:64]: yes, I understood it was a Htt-specific change.

File: PhysicsTools/Heppy/python/physicsutils/JetReCalibrator.py (hunk 1 above)
Comment (jpata) [reply PhysicsTools/Heppy/python/physicsutils/JetReCalibrator.py:112]: OK, I'll remove it.

**Comment by jpata** [comment]
I wonder if we should simply check out the heppy-side version of the `JetReCalibrator` and not deal with these merge changes.

**Comment by arizzi** [comment]
@gkasieczka @veelken can you comment? (i.e. is it ok for you if we just overwrite those changes in the merge with colin branch and then you reimplement on top if you still need?)

**Comment by gkasieczka** [comment]
Yes, that works for me and should be the easiest.

**Comment by arizzi** [comment]
@jpata ok, then please just fix the delat!=0 thing (if we agree on it)

**Comment by veelken** [comment]
I agree too.

**Comment by jpata** [comment]
done

**Comment by arizzi** [comment]
mhh ...  I merged but I see if delta!=0 is still there!

**Pull request merged**